mantidproject/mantid
# Add exclusive bounds to ArrayBoundedValidator

This PR tries to bring `ArrayBoundedValidator` more consistent with `BoundedValidator`. This allows for a new constructor in Python enabling one to write
```python
validator = IntArrayValidator(lower=0.)
```

This PR also updates some algorithms to use the new feature.


**To test:**

Try out the new `ArrayBoundedValidator` constructor with named parameters: `v = FloatArrayBoundedValidator(lower=-0.3, upper=0.7, True)` and the new methods from Python.

Check also that this does not break the old behavior: `v = IntArrayBoundedValidator(lowerBound=4, upperBound=8)` should still work.

Fixes #9777. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
